src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -147,15 +145,90 @@ int goto_instrument_parse_optionst::doit()
     get_goto_program();
     instrument_goto_program();
 
-    if(cmdline.isset("unwind"))
     {
+      bool unwind=cmdline.isset("unwind");
+      bool unwindset=cmdline.isset("unwindset");
+      bool unwindset_file=cmdline.isset("unwindset-file");
 
-      int k = stoi(cmdline.get_value("unwind"));
+      if(unwindset && unwindset_file)
+        throw "only one of --unwindset and --unwindset-file supported at a "
+              "time";
 
-      goto_unwind(goto_functions, k);
+      if(unwind || unwindset || unwindset_file)
+      {
+        int k=-1;
 
-      goto_functions.update();
-      goto_functions.compute_loop_numbers();
+        if(unwind)
+          k=safe_string2int (cmdline.get_value("unwind"));
+
+        unwind_sett unwind_set;
+
+        if(unwindset_file)
+        {
+          std::string us;
+          std::string fn=cmdline.get_value("unwindset-file");
+          std::ifstream file(fn);
```

Comment:
This unfortunately needs to include widen() on WIN32, or Unicode filenames won't work.